Feature/home screen

marina-ivanovic opened the pull request:
Changed designs of event/offering cards, added location and date attributes to the Event class, changed from tabs for displaying the home page to segmented button.

milicaradicc commented on EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopEventsFragment.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    // Method to simulate getting top event data
+    private List<Event> getTopEvents() {
```
Add some error handling or empty state management. For example, if the list of events is empty, you can display a "No Events Available" message or some fallback UI

milicaradicc commented on EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopOfferingsFragment.java:
```diff
@@ -0,0 +1,64 @@
+        recyclerView = rootView.findViewById(R.id.topOfferingsRecyclerView);
+
+        // Initialize offeringList (you will replace this with actual data fetching logic)
+        offeringList = getOfferingList();  // Ensure this method provides data
```
Add some error handling or empty state management

milicaradicc replied:
Design looks great,  implementing an empty state is necessary. Add error handling and loading states for a better user experience.

milicaradicc commented:
Design looks great, implementing an empty state is necessary. Add error handling and loading states for a better user experience.

marina-ivanovic commented:
Changes made:
- Made a display text for when the list of events/offerings is empty.
- Combined all of the xml and Java files for display of cards (allEvents/topEvents/allOfferings/topOfferings fragments and xml files) into one fragment and created the logic for displaying and hiding components depending on the tab the user clicked on.
- Changed the search bar from custom made one, to material design search bar
- Renamed the filter components, but the components themselves will be changed on another branch

- There is no longer need to separate the recycler view from the component it's getting used in, if you want to reuse the component, there is no need to make another xml, just make the recycler view for that list. The logic for putting the the cards in the recycler view is in the home fragment, and the adapters for events/offerings can be used for binding the data to the cards.
